core/src/test/scala/org/apache/spark/util/UninterruptibleThreadSuite.scala

```diff
@@ -115,6 +116,51 @@ class UninterruptibleThreadSuite extends SparkFunSuite {
     assert(interruptStatusBeforeExit)
   }
 
+  test("no runUninterruptibly") {
+    @volatile var hasInterruptedException = false
+    val t = new UninterruptibleThread("test") {
+      override def run(): Unit = {
+        if (sleep(0)) {
+          hasInterruptedException = true
+        }
+      }
+    }
+    t.interrupt()
+    t.start()
+    t.join()
+    assert(hasInterruptedException === true)
+  }
+
+  private class InterruptibleChannel(latch: CountDownLatch, lock: Object, thread: Thread)
+    extends AbstractInterruptibleChannel {
+    begin()
+    latch.countDown()
+    close()
+    override def implCloseChannel(): Unit = {
+      try {
+        Thread.sleep(Long.MaxValue)
+      } catch {
+        case _: InterruptedException => thread.interrupt()
+      }
+    }
+  }
+
+  test("uninterruptibleLock deadlock") {
+    val lock = new Object
+    val latch = new CountDownLatch(1)
+    val task = new UninterruptibleThread("task thread") {
+      override def run(): Unit = {
+        new InterruptibleChannel(latch, lock, Thread.currentThread())
```

Comment:
is this also what happened to the reported Parquet bug? To me, this looks like a weird case that the "uninterruptible" function interrupts the thread itself.

Reply:
@cloud-fan It is exactly the same deadlock as what is reported during Parquet upgrade.